hera_cal/reflections.py

```diff
@@ -267,25 +239,445 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+            edgecut_low : int
+                Nbins to flag but not window at low-side of band.
+            edgecut_hi : int
+                Nbins to flag but not window at high-side of band.
+            verbose : bool
+                If True, report feedback to stdout.
+        """
+        # get keys
+        if keys is None:
+            keys = self.pcomp_model.keys()
+
+        if not hasattr(self, 'pcomp_model_fft'):
+            self.pcomp_model_fft = DataContainer({})
+        if not hasattr(self, 'data_pmodel_resid'):
+            self.data_pmodel_resid = DataContainer({})
+
+        # get data
+        if data is None:
+            data = self.data
+
+        # iterate over keys
+        for k in keys:
+            if k in self.pcomp_model_fft and not overwrite:
+                echo("{} in pcomp_model_fft and overwrite==False, skipping...".format(k), verbose=verbose)
+                continue
+
+            # get fft of model
+            model = self.pcomp_model[k]
+
+            # ifftshift
+            if ifftshift:
+                model = np.fft.ifftshift(model, axes=-1)
+
+            # ifft to get to data space
+            if ifft:
+                model_fft = np.fft.ifft(model, axis=-1)
+            else:
+                model_fft = np.fft.fft(model, axis=-1)
+
+            # divide by a window
+            win = dspec.gen_window(window, model_fft.shape[1], alpha=alpha,
+                                   edgecut_low=edgecut_low, edgecut_hi=edgecut_hi)
+            model_fft = np.true_divide(model_fft, win, where=~np.isclose(win, 0.0))
+
+            # subtract from data
+            self.pcomp_model_fft[k] = model_fft
+            self.data_pmodel_resid[k] = data[k] - model_fft
+
+    def interp_u(self, u, times, full_times=None, uflags=None, keys=None, overwrite=False, verbose=True,
+                 mode='gpr', gp_len=600, gp_nl=0.1, optimizer=None):
+        """
+        Interpolate u modes along time, inserts into self.umode_interp
+
+        Args:
+            u : DataContainer
+                u-mode container to interpolate
+            times : 1D array
+                Holds time_array of input u modes.
+            full_times : 1D array
+                time_array to interpolate onto. Default is times.
+            uflags : DataContainer
+                Object to pull target u flags from. Default is None.
+            keys : list of tuples
+                List of baseline-pol DataContainer tuples to operate on.
+            overwrite : bool
+                If True, overwrite output data if it exists.
+            verbose : bool
+                If True, report feedback to stdout.
+            mode : str
+                Interpolation mode. Options=['gpr']
+            gp_len : length-scale of GPR in units of times
+            gp_nl : GPR noise-level in units of input u.
+        """
+        if not hasattr(self, 'umode_interp'):
+            self.umode_interp = DataContainer({})
```

Comment:
Is DataContainer the right format for this? Like, if you give it reversed keys, do you want the complex conjugate? It's no longer nTimes x nFreqs or nTimes x nDelays, so I'm not sure if you want this to be a DataContainer as opposed to a dictionary.

Reply:
Yes, we do want all of the capabilities of `DataContainer` including automatic conjugation for these products